### PR TITLE
Fix the FiveM command

### PR DESCRIPTION
**Pull request by Doges**
It was pushing errors sometimes because some people instead of setting the banner var with an image, they were setting a discord invite or a description

image

### CODE REVIEW

File: src/commands/games/fivem.js [hunk 1]
```diff
@@ -14,39 +14,45 @@ module.exports = class FiveM extends Command {
...
+          .addField(t('commands:fivem.players'), `${server.players.length}/${server.sv_maxclients}`, true)
+          .setColor(Constants.FIVEM_COLOR)
+          .addField(t('commands:fivem.map'), server.mapname || t('commands:fivem.noMap'), true)
+          .addField(t('commands:fivem.gameMode'), server.gametype || t('commands:fivem.noGameMode'), true)
+          .addField(t('commands:fivem.gameName'), server.vars.gamename || t('commands:fivem.noGameName'), true)
 
-      if (server.vars.banner_detail || server.vars.banner_connecting) {
-        embed.setImage(server.vars.banner_detail ? server.vars.banner_detail : (server.vars.banner_connecting ? server.vars.banner_connecting : null))
-      }
+        if (server.vars.banner_detail.endsWith('png' || 'jpg' || 'gif') || server.vars.banner_connecting.endsWith('png' || 'jpg' || 'gif')) {
```
Comment (almeidx) [inline src/commands/games/fivem.js:41]: This doesn't do what you expect it does. `'png' || 'jpg' || 'gif'` will evaluate to `'png'` 

```suggestion
        const allowedExtensions = ['png', 'jpg', 'gif']
        if (allowedExtensions.some(e => server.vars.banner_detail.endsWith(e)) || allowedExtensions.some(e => server.vars.banner_connecting.endsWith(e))) {
```

File: src/commands/games/fivem.js [hunk 2]
```diff
@@ -14,39 +14,45 @@ module.exports = class FiveM extends Command {
...
+          embed.setImage(server.vars.banner_detail ? server.vars.banner_detail : (server.vars.banner_connecting ? server.vars.banner_connecting : null))
+        }
 
-      channel.send(embed).then(channel.stopTyping())
+        channel.send(embed).then(channel.stopTyping())
+      }
+    } catch (e) {
```
Comment (almeidx) [inline src/commands/games/fivem.js:47]:
```suggestion
    } catch {

```

File: src/commands/games/fivem.js [hunk 3]
```diff
@@ -14,39 +14,45 @@ module.exports = class FiveM extends Command {
...
 
   getAllServers () {
     return new Promise(function (resolve, reject) {
-      fetch('http://servers-live.fivem.net/api/servers/')
-        .then(res => resolve(res.json()))
+      axios.get('http://servers-live.fivem.net/api/servers/')
+        .then(res => resolve(res.data))
         .catch(error => reject(error))
     })
   }
```
Comment (almeidx) [inline src/commands/games/fivem.js:58]: Can't suggest changes on deleted lines so, here it goes:

```js
  async getAllServers () {
    const result = await axios.get('http://servers-live.fivem.net/api/servers/')
    return result.data
  }
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/games/fivem.js (hunk 3 above)
Comment (Doges) [reply src/commands/games/fivem.js:58]: forgor about this one

File: src/commands/games/fivem.js [hunk 4]
```diff
@@ -14,40 +14,44 @@ module.exports = class FiveM extends Command {
...
-      if (server.vars.banner_detail || server.vars.banner_connecting) {
-        embed.setImage(server.vars.banner_detail ? server.vars.banner_detail : (server.vars.banner_connecting ? server.vars.banner_connecting : null))
-      }
+        const allowedExtensions = ['png', 'jpg', 'gif']
+        if (allowedExtensions.some(e => server.vars.banner_detail.endsWith(e)) || allowedExtensions.some(e => server.vars.banner_connecting.endsWith(e))) {
+          embed.setImage(server.vars.banner_detail ? server.vars.banner_detail : (server.vars.banner_connecting ? server.vars.banner_connecting : null))
+        }
 
-      channel.send(embed).then(channel.stopTyping())
+        channel.send(embed).then(channel.stopTyping())
```
Comment (almeidx) [inline src/commands/games/fivem.js:46]:
```suggestion
        channel.send(embed).then(() => channel.stopTyping())

```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]

**Comment by almeidx** [comment]
oopsie daisy

**Review by almeidx — APPROVED** [review]

**Review by zMigueel — APPROVED** [review]
LGTM

File: src/commands/games/fivem.js [hunk 5]
```diff
@@ -14,40 +14,44 @@ module.exports = class FiveM extends Command {
...
 
-  getAllServers () {
-    return new Promise(function (resolve, reject) {
-      fetch('http://servers-live.fivem.net/api/servers/')
-        .then(res => resolve(res.json()))
-        .catch(error => reject(error))
-    })
+  async getAllServers () {
+    const result = await axios.get('http://servers-live.fivem.net/api/servers/')
+    return result.data
```
Comment (metehus) [inline src/commands/games/fivem.js:55]:
```js
return axios.get('http://servers-live.fivem.net/api/servers/').then(res => res.data)
```

**Review by metehus — CHANGES_REQUESTED** [review]
hihi

File: src/commands/games/fivem.js (hunk 5 above)
Comment (Doges) [reply src/commands/games/fivem.js:55]: faz uma sugestão filja daputa

Comment (almeidx) [reply src/commands/games/fivem.js:55]: If that's what you're going with then make the function *not* async

File: src/commands/games/fivem.js [hunk 6]
```diff
@@ -14,40 +14,43 @@ module.exports = class FiveM extends Command {
...
     }
   }
 
-  getAllServers () {
-    return new Promise(function (resolve, reject) {
-      fetch('http://servers-live.fivem.net/api/servers/')
-        .then(res => resolve(res.json()))
-        .catch(error => reject(error))
-    })
+  async getAllServers () {
```
Comment (almeidx) [inline src/commands/games/fivem.js:53]:
```suggestion
  getAllServers () {
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/games/fivem.js (hunk 6 above)
Comment (Doges) [reply src/commands/games/fivem.js:53]: ae @metehus aprende com ele a fazer sugestão 😎

**Review by almeidx — APPROVED** [review]

File: src/commands/games/fivem.js (hunk 6 above)
Comment (metehus) [reply src/commands/games/fivem.js]: to no celular

Comment (Doges) [reply src/commands/games/fivem.js]: aporra

**Review by moruzerinho6 — APPROVED** [review]

File: src/commands/games/fivem.js [hunk 7]
```diff
@@ -14,40 +14,43 @@ module.exports = class FiveM extends Command {
...
-        throw new CommandError(`${t('commands:fivem.serverUnreachableTitle')} ${t('commands:fivem.serverUnreachableDescription')}`)
-      })
+    try {
+      const testregex = /\^[0-9]/g
+      const embed = new SwitchbladeEmbed(author)
+      channel.startTyping()
+      const id = address.replace(/(http(s?):\/\/)?(cfx.re\/join\/)/g, '')
+      const server = await axios(`https://servers-frontend.fivem.net/api/servers/single/${id}`)
+        .then(res => res.data)
+        .then(data => data.Data)
```
Comment (metehus) [inline src/commands/games/fivem.js:24]:
```js
.then(res => res.data.Data)
```

**Review by metehus — CHANGES_REQUESTED** [review]

File: src/commands/games/fivem.js (hunk 7 above)
Comment (Doges) [reply src/commands/games/fivem.js:24]: vai se foder

**Review by moruzerinho6 — APPROVED** [review]
Cringe

**Review by almeidx — APPROVED** [review]
Not a particular fan of this code but won't block

**Comment by Doges** [comment]
> 
> 
> Cringe



> 
> 
> Not a particular fan of this code but won't block

https://user-images.githubusercontent.com/31023616/135768924-45e172a8-f1ea-4765-8202-356882e3fb50.mp4

**Review by metehus — APPROVED** [review]

**Pull request merged**